ctest-next/src/template.rs

```diff
@@ -291,4 +458,139 @@ impl<'a> TranslateHelper<'a> {
 
         Ok(self.generator.map(item))
     }
+
+    /// Recursively get the properly mapped type for some syn::Type.
+    ///
+    /// This method is meant to only be used to translate simpler types, such as primitives or
+    /// pointers/references to primitives.
+    fn c_type_raw(&self, ty: &syn::Type) -> Result<String, TranslationError> {
+        let type_name = match ty {
+            syn::Type::Path(p) => p.path.segments.last().unwrap().ident.to_string(),
+            syn::Type::Ptr(p) => self.c_type_raw(&p.elem)?,
+            syn::Type::Reference(r) => self.c_type_raw(&r.elem)?,
+            _ => ty.to_token_stream().to_string(),
+        };
+
+        let unmapped_c_type = self.translator.translate_type(ty)?;
+        let item = if self.ffi_items.contains_struct(&type_name) {
+            MapInput::StructType(&unmapped_c_type)
+        } else if self.ffi_items.contains_union(&type_name) {
+            MapInput::UnionType(&unmapped_c_type)
+        } else {
+            MapInput::Type(&unmapped_c_type)
+        };
+
+        Ok(self.generator.map(item))
```

Comment:
If you get the chance, `map` could use a more descriptive name. Maybe `rty_to_cty`?